easybuild/easyblocks/p/perl.py

```diff
@@ -42,17 +43,30 @@
 class EB_Perl(ConfigureMake):
     """Support for building and installing Perl."""
 
+    @staticmethod
+    def extra_options():
+        """Add extra config options specific to Perl."""
+        extra_vars = {
+            'use_perl_threads': [True, "Use internal Perl threads by means of the -Dusethreads compiler directive", CUSTOM],
+        }
+        return ConfigureMake.extra_options(extra_vars)
+
     def configure_step(self):
         """
         Configure Perl build: run ./Configure instead of ./configure with some different options
         """
         configopts = ' '.join([
             self.cfg['configopts'],
-            "-Dusethreads",
             '-Dcc="%s"' % os.getenv('CC'),
             '-Dccflags="%s"' % os.getenv('CFLAGS'),
             '-Dinc_version_list=none',
         ])
+        if self.cfg['use_perl_threads']:
+            configopts = ' '.join([
+                self.cfg['configopts'],
+                '-Dusethreads',
+            ])
+
         cmd = './Configure -de %s -Dprefix="%s" ' % (configopts, self.installdir)
```

Comment:
@valtandor: I would clean this up a bit as follows, i.e. construct `configopts` as a list first, and only `join` when it's used:

```python
configopts = [
...
]
if self.cfg['use_perl_threads']:
    configopts.append('-Dusethreads')

cmd = './Configure -de %s -Dprefix="%s" ' % (' '.join(configopts), self.installdir)
```